parquet-common/src/main/java/org/apache/parquet/hadoop/metadata/ColumnPath.java

```diff
@@ -41,7 +47,14 @@ protected ColumnPath toCanonical(ColumnPath value) {
 
   public static ColumnPath fromDotString(String path) {
     checkNotNull(path, "path");
-    return get(path.split("\\."));
+    String[] parts = path.split("\\.(?=([^`]*`[^`]*`)*[^`]*$)");
```

Comment:
I think we should centralize the quote parsing logic.
This is the 3rd place where we unquote/unescape quoted names.